oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/event/Event.java

```diff
@@ -0,0 +1,231 @@
+
+@Getter
+@Setter
+@ScopeDeclaration(id = EVENT, name = "Event")
+@Stream(name = Event.INDEX_NAME, scopeId = EVENT, builder = Event.Builder.class, processor = MetricsStreamProcessor.class)
+@EqualsAndHashCode(
+    callSuper = false,
+    of = "uuid"
+)
+@MetricsExtension(supportDownSampling = false, supportUpdate = true)
+public class Event extends Metrics {
+
+    public static final String INDEX_NAME = "events";
+
+    public static final String UUID = "uuid";
+
+    public static final String SERVICE = "service";
+
+    public static final String SERVICE_INSTANCE = "service_instance";
+
+    public static final String ENDPOINT = "endpoint";
+
+    public static final String NAME = "name";
+
+    public static final String TYPE = "type";
+
+    public static final String MESSAGE = "message";
+
+    public static final String PARAMETERS = "parameters";
+
+    public static final String START_TIME = "start_time";
+
+    public static final String END_TIME = "end_time";
+
+    @Override
+    public String id() {
+        return getUuid();
+    }
+
+    @Column(columnName = UUID)
+    private String uuid;
+
+    @Column(columnName = SERVICE)
+    private String service;
+
+    @Column(columnName = SERVICE_INSTANCE)
+    private String serviceInstance;
+
+    @Column(columnName = ENDPOINT)
+    private String endpoint;
+
+    @Column(columnName = NAME)
+    private String name;
+
+    @Column(columnName = TYPE)
+    private String type;
+
+    @Column(columnName = MESSAGE)
+    private String message;
+
+    @Column(columnName = PARAMETERS, storageOnly = true)
+    private String parameters;
+
+    @Column(columnName = START_TIME)
+    private long startTime;
+
+    @Column(columnName = END_TIME)
+    private long endTime;
+
+    @Override
+    public boolean combine(final Metrics metrics) {
+        final Event event = (Event) metrics;
+
+        // Set time bucket only when it's never set.
+        if (getTimeBucket() <= 0) {
+            if (event.getStartTime() > 0) {
+                setTimeBucket(TimeBucket.getMinuteTimeBucket(event.getStartTime()));
+            } else if (event.getEndTime() > 0) {
+                setTimeBucket(TimeBucket.getMinuteTimeBucket(event.getEndTime()));
+            }
+        }
+
+        // Set start time only when it's never set, (`start` event may come after `end` event).
+        if (getStartTime() <= 0 && event.getStartTime() > 0) {
+            setStartTime(event.getStartTime());
+        }
+
+        if (event.getEndTime() > 0) {
+            setEndTime(event.getEndTime());
+        }
+
+        if (StringUtil.isNotBlank(event.getType())) {
+            setType(event.getType());
+        }
+        if (StringUtil.isNotBlank(event.getMessage())) {
+            setType(event.getMessage());
+        }
+        if (StringUtil.isNotBlank(event.getParameters())) {
+            setParameters(event.getParameters());
+        }
+        return true;
```

Comment:
So, we still always return `true`, right? timebucket will be there always.

Reply:
Yes